docs/kb/semgrep-code/semgrep-scan-troubleshooting.md

```diff
@@ -0,0 +1,56 @@
+---
+tags:
+  - Semgrep Supply Chain
+  - Semgrep Code
+description: Troubleshoot common issues with Semgrep scans.
+---
+
+# A Semgrep scan is having a problem - what next?
+
+If a Semgrep scan is failing or running slowly, try the following steps to investigate:
+
+1. [Update Semgrep](/docs/upgrading/) to the latest version, if you are not currently running the latest version. Some errors result from an older version of Semgrep being used with newer rules.
+2. Re-run the scan with either the `-v`/`--verbose` or `--debug` (extremely verbose) flags. These options provide more information about what is failing.
+3. Running Semgrep Pro Engine in the scan? Remove any options starting with `--pro`, or run Semgrep with `--oss-only`. This allows isolation of any issues related to Semgrep Pro Engine, and often speeds up a scan or reduces memory usage.
```

Comment:
```suggestion
3. If you are running Semgrep Pro Engine in the scan, remove any options starting with `--pro`, or run Semgrep with `--oss-only`. This allows isolation of any issues related to Semgrep Pro Engine, and often speeds up a scan or reduces memory usage.
```
Just concerned that non-native English readers will be confused by the gerund ending in a question.